Source/ARTPushChannelSubscriptions.m

```diff
@@ -55,6 +58,7 @@ - (void)save:(ARTPushChannelSubscription *)channelSubscription callback:(void (^
     request.HTTPMethod = @"POST";
     request.HTTPBody = [[_rest defaultEncoder] encodePushChannelSubscription:channelSubscription error:nil];
     [request setValue:[[_rest defaultEncoder] mimeType] forHTTPHeaderField:@"Content-Type"];
+    [request setDeviceAuthentication:channelSubscription.deviceId localDevice:local];
```

Comment:
Why not using `_rest.device` directly here and drop the `local ` variable?

Reply:
Because that will lead to a deadlock. `_rest.device` is synced using `_queue `.